subprojects/gradle/utils/src/main/kotlin/com/avito/utils/gradle/EnvArgs.kt

```diff
@@ -17,61 +20,118 @@ val Project.envArgs by ProjectProperty.lazy<EnvArgs>(scope = ROOT_PROJECT) { pro
 
 interface EnvArgs {
 
+    sealed class Build : Serializable {
+        /**
+         * @todo change to string after refactor [TestRunEnvironment.teamcityBuildId]
+         */
+        abstract val id: Int
+        abstract val url: String
+        abstract val number: String
+        abstract val type: String
+
+        internal class Local(id: Id) : Build() {
+            override val id = id.id
+            override val url = "No url. This is local build"
+            override val number = "local"
+            override val type = "${userName}-local-${TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis())}"
```

Comment:
Do we need a unique type on every launch?
How it correlates with a local report? If I run tests on the same commit, will I reuse the same report?

Reply:
It correlates with k8s deployment name

Reply:
Report use git info. In teamcity every run has unique build_id.